kernel/src/actions/visitors.rs

```diff
@@ -724,4 +799,110 @@ mod tests {
             })
         );
     }
+
+    #[test]
+    fn test_parse_domain_metadata() {
+        let json_strings: StringArray = vec![
+            r#"{"metaData":{"id":"aff5cb91-8cd9-4195-aef9-446908507302","format":{"provider":"parquet","options":{}},"schemaString":"{\"type\":\"struct\",\"fields\":[{\"name\":\"c1\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"c2\",\"type\":\"string\",\"nullable\":true,\"metadata\":{}},{\"name\":\"c3\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}]}","partitionColumns":["c1","c2"],"configuration":{},"createdTime":1670892997849}}"#,
+            r#"{"domainMetadata":{"domain": "zach1","configuration":"cfg1","removed": true}}"#,
+            r#"{"domainMetadata":{"domain": "zach2","configuration":"cfg2","removed": false}}"#,
+            r#"{"domainMetadata":{"domain": "zach3","configuration":"cfg3","removed": true}}"#,
+            r#"{"domainMetadata":{"domain": "zach4","configuration":"cfg4","removed": false}}"#,
+            r#"{"domainMetadata":{"domain": "zach5","configuration":"cfg5","removed": true}}"#,
+            r#"{"domainMetadata":{"domain": "zach6","configuration":"cfg6","removed": false}}"#,
+        ]
+        .into();
+        let batch1 = parse_json_batch(json_strings);
+        let json_strings: StringArray = vec![
+            r#"{"metaData":{"id":"aff5cb91-8cd9-4195-aef9-446908507302","format":{"provider":"parquet","options":{}},"schemaString":"{\"type\":\"struct\",\"fields\":[{\"name\":\"c1\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}},{\"name\":\"c2\",\"type\":\"string\",\"nullable\":true,\"metadata\":{}},{\"name\":\"c3\",\"type\":\"integer\",\"nullable\":true,\"metadata\":{}}]}","partitionColumns":["c1","c2"],"configuration":{},"createdTime":1670892997849}}"#,
+            r#"{"domainMetadata":{"domain": "zach1","configuration":"old_cfg1","removed": true}}"#,
```

Comment:
This one should _not_ be `removed` to do a proper test. I think we have to remember that the domain has been removed by a later action and ignore an earlier "not removed" one. I.e. from the protocol:
> [domain metadata] actions with removed=true act as tombstones to suppress earlier versions.

Reply:
yup thanks for catching this - i probably implemented this a little too fast..

this brings up an interesting scenario though: if we have a `domainMetadata { domain1, config0, removed=true }` in commit 0, followed by a `domainMetadata { domain1, config1, removed=false }` in commit 1, we would return `"config1"` to the user when they ask for `domain1` at version 1?

Reply:
got it- thanks! i've implemented this now :)